zulip_bots/zulip_bots/test_lib.py

```diff
@@ -136,12 +136,9 @@ def mock_http_conversation(self, test_name):
             http_response = http_data.get('response')
             http_headers = http_data.get('response-headers')
             with patch('requests.get') as mock_get:
-                mock_result = mock.MagicMock()
-                mock_result.json.return_value = http_response
-                if 'text' in http_response:
-                    mock_result.text = http_response.get('text', None)
+                mock_result = requests.Response()
```

Comment:
I would explain with a comment how it's safe to use a "real" `Response` object in a `mock_result` context.